modules/ding_ddbasic/modules/ding_ddbasic_opening_hours/ding_ddbasic_opening_hours.theme.inc

```diff
@@ -0,0 +1,264 @@
+  }
+
+  // Get and sort all the instances in the given timespan.
+  $instances = opening_hours_instance_load_multiple(
+    array($variables['node']->nid),
+    date('Y-m-d', $timespan[0]),
+    date('Y-m-d', $timespan[1])
+  );
+
+  // Add closed days.
+  for ($day = $timespan[0]; $day <= $timespan[1]; $day += DING_DDBASIC_OPENING_HOURS_ONE_DAY) {
+    $date = date('Y-m-d', $day);
+    $closed = TRUE;
+    foreach ($instances as $instance) {
+      if ($instance->date == $date) {
+        $closed = FALSE;
+      }
+    }
+
+    if ($closed === TRUE) {
+      $instances[] = (object) array(
+        'instance_id' => -1,
+        'nid' => $variables['node']->nid,
+        'start_time' => '0',
+        'end_time' => '0',
+        'category_tid' => NULL,
+        'date' => $date,
+        'closed' => TRUE,
+        'notice' => t('Closed'),
+      );
+    }
+  }
+
+  usort($instances, 'ding_ddbasic_opening_hours_sort');
+
+  // Add NULL to the categories, if the the NO_CATEGORY category is selected.
+  // So an empty category_tid of the instance matches the selected categories.
+  if (in_array(DING_DDBASIC_OPENING_HOURS_NO_CATEGORY, $variables['categories'])) {
+    $variables['categories'][''] = NULL;
+  }
+
+  // Insert the instances into the structured array.
+  foreach ($instances as $instance) {
+    if (!empty($variables['categories']) && !in_array($instance->category_tid, $variables['categories'])) {
+      continue;
+    }
+
+    $category_weight = ding_ddbasic_opening_hours_get_category_weight($instance->category_tid);
+    $time = strtotime($instance->date);
+    $day = format_date($time, 'custom', 'l');
+
+    if (!isset($structured[$day])) {
+      $structured[$day] = array(
+        'cols' => array(),
+        'extra' => array(),
+      );
+
+      if ($time == $today) {
+        $structured[$day]['extra']['class'] = array('today');
+      }
+    }
+
+    if (!empty($instance->notice)) {
+      $notice = '<span class="notice">' . $instance->notice . '</span>';
+    }
+    else {
+      $notice = '';
+    }
+
+    if (empty($instance->closed)) {
+      $structured[$day]['cols'][$category_weight] = t('@from - @to', array(
+        '@from' => $instance->start_time,
+        '@to' => $instance->end_time,
+      )) . $notice;
+    }
+    else {
+      $structured[$day]['cols'][$category_weight] = $notice;
+    }
+
+    if (!isset($categories[$category_weight])) {
+      $categories[$category_weight] = ding_ddbasic_opening_hours_get_category_name($instance->category_tid);
+    }
+  }
+
+  // Sort the categories by key (weight).
+  ksort($categories);
+
+  $title = t('Week @week, @from - @to', array(
+    '@week' => format_date($timespan[0], 'custom', 'W'),
+    '@from' => format_date($timespan[0], 'custom', 'd.m'),
+    '@to' => format_date($timespan[1], 'custom', 'd.m'),
+  ));
+
+  $variables['table'] = ding_ddbasic_opening_hours_table(
+    $title,
+    $categories,
+    $structured
+  );
+
+  drupal_add_library('system', 'drupal.ajax');
```

Comment:
Drupal only includes the drupal.ajax library if it's needed, and we can't assume there's anything else on the page that has included the library.